send modern-style ISO19115:2003 md links to geoserver when geopublishing

landryb opened the pull request:
this should please mapstore.

fixes #195

landryb committed:
send modern-style ISO19115:2003 md links to geoserver when geopublishing

this should please mapstore.

pmauduit reviewed (approved):
LGTM, I guess the CSW query was to be more versatile (no matter the catalogue in used), but since we are obviously dealing with a GN instance here, I don't mind :)

pmauduit commented:
the failing build sounds weird to me though: https://github.com/georchestra/geonetwork/runs/5156370084?check_suite_focus=true it complains about missing jars coming from georchestra/georchestra, but I thought we fixed this some time ago.

landryb commented:
> LGTM, I guess the CSW query was to be more versatile (no matter the catalogue in used), but since we are obviously dealing with a GN instance here, I don't mind :)

problem with the csw query was also that it doesnt please mapstore which expects a md document (eg starting with `gmd:MD_Metadata`) and not a csw response (starting with `csw:GetRecordByIdResponse`), as i figured out the hard way :D

landryb commented:
> the failing build sounds weird to me though: https://github.com/georchestra/geonetwork/runs/5156370084?check_suite_focus=true it complains about missing jars coming from georchestra/georchestra, but I thought we fixed this some time ago.

fixed for the makefile build from georchestra repo, yeah: georchestra/georchestra@0df0e293c0

But i dont think the underlying issue of gn4 build not properly depending on some georchestra jars in some maven plumbing has been fixed.

the build log complains:
```
Warning: 532 [WARNING] The POM for org.georchestra:georchestra-testcontainers:jar:21.0-SNAPSHOT is missing, no dependency information available
```
